doc/cmis-module-enhancement/cmis-module-enhancement.md

```diff
@@ -0,0 +1,224 @@
+This event shall be used by Ports OA as a trigger for enabling or disabling the host_tx_signal (using setting true/false to SAI_PORT_ATTR_HOST_TX_SIGNAL_ENABLE on the Port Object).
+
+To detect this insertion/removal event the Ports Orch Agent shall register for changes in the TRANSCIEVER_TABLE in STATE DB and will do the following:
+* INSERTION event – set SAI_PORT_ATTR_HOST_TX_SIGNAL_ENABLE to TRUE
+* REMOVAL event   - set SAI_PORT_ATTR_HOST_TX_SIGNAL_ENABLE to FALSE
+
+Please note that on platforms supporting this functionality, SDK/FW shall start the transmission of high-speed signal from ASIC to a module only if both criteria below are met:
+* TX signal is enabled (via setting SAI_PORT_ATTR_HOST_TX_SIGNAL_ENABLE to TRUE)
+* Admin Status is set to UP
+
+Whenever one of them is set to false (e.g. TX signal is disabled due to removal event of Admin station is set to DOWN), the SDK/FW stops transmission of this signal.
+
+This approach allows to avoid the issue of unterminated transmission when ASIC starts the transmission of the high-speed signal even when no module is plugged-in causing cross-talks to adjacent ports, high EMI/BER and shortening the transceiver lifespan
+
+On port creation for platforms supporting proposed enhancements, the SAI_PORT_ATTR_HOST_TX_SIGNAL_ENABLE shall be explicitly set by Ports OA to FALSE since its default SAI value is defined as TRUE for backward-compatibility reasons.
+
+#### 6.3.3. Handling of HW-based “host_tx_ready” event
+Per description above, on the platform supporting new enhancements, the SDK/FW shall send an asynchronous notification of “host_tx_ready” to SWSS when the ASIC starts/stops sending high-speed signal to a module.
+
+The vendor SDK shall internally support this indication from ASIC (via the trap mechanism) and shall call the notification callback registered by Ports OA on initialization .
+The backward-compatibility is supported by a new design. Therefore, the handling shall be as follows:
+
+| Platform Type                            | Ports OA Behavior                                                                                                                                            |   |
+|------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------|---|
+| Platform supporting new enhancements     | On return from SAI call of setting port's Admin status, nothing is done, unless the notification event "host_tx_ready" is received and consumed by Ports OA. |   |
+| Platform not supporting new enhancements | Ports OA will set "host_tx_ready" in STATE DB right after Admin status SAI call returns.<br>(UP -> "true", DOWN -> "false")                                  |   |
+
+
+##### 6.3.3.1. Warm-boot handling
+After warm-boot, SONiC will not receive a new “host_tx_ready” notification from SAI (because nothing has changed in ASIC/module). 
```

Comment:
@noaOrMlnx I am not sure what is the usecase to have host_tx_ready updated during warm reboot? Xcvrd is not going to act upon this host_tx_ready notification anyways during warm reboot

Reply:
@prgeor The only reason here is to have STATE DB updated with lower layers status. 
This was also discussed during HLD review, it is not mandatory but nice to have it all updated with the same status.

Reply:
@noaOrMlnx can we preserve the STATE_DB's TRANSCEIVER_INFO table and PORT_TABLE (host_tx_ready) across warm-reboot so that it does not cause link flap.

Reply:
@prgeor Hi Prince, As far as I checked, TRANSCEIVER_INFO table does not deleted when doing warm-reboot. 
only PORT_TABLE.
For PORT_TABLE, please check my above comment, which explains why I think it will not be the solution.

Reply:
@noaOrMlnx it seems except few TABLES mentioned [here](https://github.com/sonic-net/sonic-utilities/blob/master/scripts/fast-reboot#L247) other tables get deleted before warm-reboot

Reply:
@noaOrMlnx is your confirmation verified on a real switch? i.e if on warm-reboot `host_tx_ready` does not get deleted from STATE_DB and hence does not need to be preserved and restored before and after warm-reboot?

Reply:
my bad, both tables are deleted during warm reboot